Catena4430_cMeasurementLoop.cpp

```diff
@@ -62,34 +62,77 @@ void cMeasurementLoop::begin()
     this->m_PelletFeeder.begin(gCatena);
 
     Wire.begin();
-    if (this->m_BME280.begin(BME280_ADDRESS, Adafruit_BME280::OPERATING_MODE::Sleep))
+
+    if (!this->flashParam())
         {
-        this->m_fBme280 = true;
+        gCatena.SafePrintf(
+                "**Unable to fetch flash parameters, assuming 4610 version 1 (rev C or earlier)!\n"
+                );
+        this->boardRev = 2;
         }
-    else
-        {
-        this->m_fBme280 = false;
-        gCatena.SafePrintf("No BME280 found: check wiring\n");
+    else {
+        this->printBoardInfo();
         }
 
-    if (this->m_si1133.begin())
+    if (!this->isVersion2())
         {
-        this->m_fSi1133 = true;
-        this->m_fLowLight = true;
+        kMessageFormat = MeasurementFormat::kMessageFormatV1;
 
-        auto const measConfig =	Catena_Si1133::ChannelConfiguration_t()
-            .setAdcMux(Catena_Si1133::InputLed_t::LargeWhite)
-            .setSwGainCode(7)
-            .setHwGainCode(4)
-            .setPostShift(1)
-            .set24bit(true);
+        if (this->m_BME280.begin(BME280_ADDRESS, Adafruit_BME280::OPERATING_MODE::Sleep))
+            {
+            this->m_fBme280 = true;
+            }
+        else
+            {
+            this->m_fBme280 = false;
+            gCatena.SafePrintf("No BME280 found: check wiring\n");
+            }
+
+        if (this->m_si1133.begin())
+            {
+            this->m_fSi1133 = true;
+            this->m_fLowLight = true;
 
-        this->m_si1133.configure(0, measConfig, 0);
+            auto const measConfig =	Catena_Si1133::ChannelConfiguration_t()
+                .setAdcMux(Catena_Si1133::InputLed_t::LargeWhite)
+                .setSwGainCode(7)
+                .setHwGainCode(4)
+                .setPostShift(1)
+                .set24bit(true);
+
+            this->m_si1133.configure(0, measConfig, 0);
+            }
+        else
+            {
+            this->m_fSi1133 = false;
+            gCatena.SafePrintf("No Si1133 found: check hardware\n");
+            }
         }
     else
         {
-        this->m_fSi1133 = false;
-        gCatena.SafePrintf("No Si1133 found: check hardware\n");
+        /* this is a version 2 board (rev D or later) */
+        kMessageFormat = MeasurementFormat::kMessageFormatV2;
+
+        if (this->m_Sht.begin())
+            {
+            this->m_fSht3x = true;
+            }
+        else
+            {
+            this->m_fSht3x = false;
+            gCatena.SafePrintf("No SHT3x found: check wiring\n");
+            }
+
+        if (this->m_Ltr.begin())
+            {
+            this->m_fHardError = false;
+            this->m_fLtr329 = true;
+            }
+        else
+            {
+            this->m_fLtr329 = false;
+            gCatena.SafePrintf("No LTR329 found: check hardware\n");
+            }
```

Comment:
See above. No references to BME etc code will appear in `setup_xxx` after the refactor.